### ISSUE TITLE
4.3.2 additional nodes hanging in a cluster on startup

### ISSUE THREAD

**Issue by mloebl** [emqx/emqx#4894]
**Environment**:

- EMQ X version (e.g. `emqx_ctl status`):  4.3.2
- Hardware configuration (e.g. `lscpu`): aarch64
- OS (e.g. `cat /etc/os-release`): Ubuntu 20.04.2 LTS
- Kernel (e.g. `uname -a`): 5.4.0-1035-raspi
- Erlang/OTP version (in case you build emqx from source code): 
- Others: k3s 1.21

**What happened and what you expected to happen**:
First node comes up fine, additional nodes when they come up hang up here:
```
log.to = "console"
node.max_ets_tables = "2097152"
node.max_ports = "1048576"
node.name = "my-emqx@my-emqx-1.my-emqx-headless.default.svc.cluster.local"
node.process_limit = "2097152"
rpc.port_discovery = "manual"
Exec: /opt/emqx/erts-11.1.8/bin/erlexec -noshell -noinput +Bd -boot /opt/emqx/releases/4.3.2/start -mode embedded -boot_var ERTS_LIB_DIR /opt/emqx/erts-11.1.8/../lib -mnesia dir "/opt/emqx/data/mnesia/my-emqx@my-emqx-1.my-emqx-headless.default.svc.cluster.local" -config /opt/emqx/data/configs/app.2021.05.28.15.42.52.config -args_file /opt/emqx/data/configs/vm.2021.05.28.15.42.52.args -vm_args /opt/emqx/data/configs/vm.2021.05.28.15.42.52.args -start_epmd false -epmd_module ekka_epmd -proto_dist ekka -- foreground
Root: /opt/emqx
Starting emqx on node my-emqx@my-emqx-1.my-emqx-headless.default.svc.cluster.local
2021-05-28T15:43:42.206927+00:00 [warning] ekka_mnesia: still waiting for table(s): [emqx_resource,mqtt_app,emqx_rule,emqx_banned,mqtt_admin]
2021-05-28T15:44:12.207941+00:00 [warning] ekka_mnesia: still waiting for table(s): [emqx_resource,mqtt_app,emqx_rule,emqx_banned,mqtt_admin]
2021-05-28T15:44:42.209061+00:00 [warning] ekka_mnesia: still waiting for table(s): [emqx_resource,mqtt_app,emqx_rule,emqx_banned,mqtt_admin]
2021-05-28T15:45:12.210893+00:00 [warning] ekka_mnesia: still waiting for table(s): [emqx_resource,mqtt_app,emqx_rule,emqx_banned,mqtt_admin]
```

**How to reproduce it (as minimally and precisely as possible)**:
Had an existing 4.3.1 3 node cluster in kubernetes. First tried a helm rolling upgrade, as well as a full redeploy of all the pods. This is come from version 4.3.1 with persistent storage. I have not tried resetting the storage. 

**Anything else we need to know?**:

This is running in a kubernetes cluster deployed via helm

**Comment by mloebl** [comment]
Update, I tried to role back to 4.3.1, did not have any luck. Had to delete persistent storage and all the nodes came up as 4.3.2

**Comment by zmstone** [comment]
Hi @mloebl 
This is a known issue of clustered nodes' restart order.
we are looking into it.

**Comment by qzhuyan** [comment]
> First tried a helm rolling upgrade, as well as a full redeploy of all the pods. 

@mloebl do you mean rolling upgrade attempt was failed first? Then you tried a redeployment? Can you specify which commands did you use for both?

**Comment by mloebl** [comment]
@qzhuyan  I did a `helm upgrade my-emqx -f ./values.yaml emqx/emqx` which had 3 nodes in the cluster. It released one node so two were still available, and it stuck with this error.  After that didn't work, I wondered if there was a 4.3.1 <-> 4.3.2 compatibility issues between nodes, so I did a `helm delete my-emqx`, followed by a `helm install my-emqx -f ./values.yaml emqx/emqx` to redeploy all of them. This didn't work until I deleted the persistent storage.
